tls/s2n_client_cert_verify.c

```diff
@@ -33,14 +33,7 @@ int s2n_client_cert_verify_recv(struct s2n_connection *conn)
 
     struct s2n_stuffer *in = &conn->handshake.io;
 
-    if (conn->actual_protocol_version < S2N_TLS12) {
-        POSIX_GUARD(s2n_choose_default_sig_scheme(conn,
```

Comment:
I know that you're not removing s2n_choose_default_sig_scheme in this PR, but you are replacing it here with s2n_signature_algorithm_recv, so I'm wondering if the tests for s2n_choose_default_sig_scheme should instead be calling s2n_signature_algorithm_recv now.

Reply:
They're not equivalent.

The reason I didn't just reuse s2n_choose_default_sig_scheme is that it does LOT more than s2n_signature_algorithm_recv actually needs. s2n_signature_algorithm_recv only cares about the defaults for <TLS1.2. All the rest of s2n_choose_default_sig_scheme is only relevant to receiving the peer's preferences, not to receiving the peer's selection.

The current s2n_choose_default_sig_scheme tests will be more relevant to the next PR, where I refactor the logic for receiving the peer's preferences.